java/core/src/main/java/com/whylogs/core/metrics/aggregators/IAggregator.java

```diff
@@ -0,0 +1,9 @@
+package com.whylogs.core.metrics.aggregators;
+
+import com.whylogs.core.errors.UnsupportedError;
+
+import java.util.function.BiFunction;
+
+public interface IAggregator<T, U, R> extends BiFunction<T, U, R> {
```

Comment:
Is it not always the case that `T == U == R` ?

Reply:
I guess I was leaving it open in case a custom one wanted to be able to merge two different types. Sheep and cows to get the herd type of thing. I can change it if we would rather.

Reply:
I don't think we (should) support merging components outside the context of metrics. I don't think we (should) support heterogeneous metric merges either :)  So I vote `T == U == R`.

Reply:
Done